### ISSUE TITLE
unit tests in vs code hanging on _startIsolate

### ISSUE THREAD

**Issue by bsutton** [dart-lang/test#1128]
I've being debugging unit tests in vscode for some time.

This morning when I tried to start a vscode unit test, the unit test hangs in the debugger.
I can debug non-unit test code in the same project without a problem.

When I start a unit test debug session in the vs code debug panel I see two isolates under the 'call stack' panel.
Both isolates show 'main' and 'running'.
This is what I normally see.

The problem this morning is that the unit test just are hanging.
If I toggle the new vscode option to allow me to debug packages and SDK code and then click the debug 'pause' option on the second 'main' isolate in the 'call stack' panel I can then click the 'step into' button.
When I do this the debugger shows 'isolate_patch.dart' line 272 which is the first line of _startIsolate.

If I then click the debug 'run' option my unit tests will run as per normal.

I've since ran a snap refresh and apt upgrade.
Neither updated any packages related to vscode/dart.

I've rebooted my PC and I'm still getting the same results.

The problem occurs whether I start from a launch script as below or if I try to directly debug the unit test by right clicking the library and selecting 'Start Debugging'.

If I right click the library and select 'Start without Debugging' the unit test runs without a problem.

```
 dart --version
Dart VM version: 2.7.0 (Unknown timestamp) on "linux_x64"
```
Vscode version
```
Version: 1.41.1
Commit: 26076a4de974ead31f97692a0d32f90d735645c0
Date: 2019-12-18T15:04:31.999Z
Electron: 6.1.5
Chrome: 76.0.3809.146
Node.js: 12.4.0
V8: 7.6.303.31-electron.0
OS: Linux x64 5.3.0-24-generic snap
```

```
test: ^1.6.0
```

**Comment by jakemac53** [comment]
Everything is working fine for me so we would need some more information to take action here, I have some questions below:

Can you create a minimal reproduction case that will cause this?

Does it happen for all your tests or just one?

What is your exact version of the `test` package? (you can run `pub deps` to print this)

What is your exact version of the Dart VsCode plugin?

**Comment by bsutton** [comment]
So it looks like I may have a corrupt state.vscdb.

I started by pulling a new copy of my project from github into a new folder and found it worked fine.

I then found that if I renamed the project folder and re-opened it in vscode that it ran just fine.
I noted that with a new folder name vscode no-longer remembered what folders I had expanded in the file explorer.

I'm also having no problems with other projects.

So I went looking for a vscode config file that was held outside of my project folder.

I found a set of files under:
~/.config/Code/User/workspaceStorage

There was a binary file under a UID folder:

state.vscdb

I renamed this file to state.vscdb.bad

relaunched my project and everything now works.

So I'm going to close this issue and leave the above notes in case someone else has a similar problem.
